kentik_api_library/kentik_api/api_connection/api_connector.py

```diff
@@ -37,15 +57,29 @@ def send(self, api_call: APICall, payload: Optional[Dict[str, Any]] = None) -> A
             response = requests.delete(url, auth=self._auth, headers=self.DEFAULT_HEADERS, json=payload)
         else:
             raise ValueError(f"Improper API call method: {api_call.method}")
+        return response
 
-        self._validate_response(response)
+    def _get_api_query_url(self, url_path: str) -> str:
+        return self._api_url + url_path
 
-        return APICallResponse(response.status_code, response.text)
+    @staticmethod
+    def _handle_error(response: requests.Response) -> None:
+        if response.status_code >= 400:
+            raise new_api_error(response.text, response.status_code)
 
-    def _get_api_query_url(self, api_method: str) -> str:
-        return self._api_url + api_method
 
-    def _validate_response(self, response: requests.Response) -> None:
-        # http error handling can be implemented here eg with exceptions, Expected[return, error] or error codes
-        if response.status_code >= 400:
-            self._logger.error("code: %d, body: %s", response.status_code, response.text)
+def new_api_error(message: str, http_status_code: int) -> KentikAPIError:
+    common_errors = {
+        HTTPStatus.BAD_REQUEST.value: BadRequestError,
+        HTTPStatus.UNAUTHORIZED.value: AuthError,
+        HTTPStatus.FORBIDDEN.value: AuthError,
+        HTTPStatus.NOT_FOUND.value: NotFoundError,
+        HTTPStatus.TOO_MANY_REQUESTS.value: RateLimitExceededError,
+        HTTPStatus.SERVICE_UNAVAILABLE.value: UnavailabilityError,
+        HTTPStatus.GATEWAY_TIMEOUT.value: UnavailabilityError,
+    }
+    try:
+        error = common_errors[http_status_code](message)
```

Comment:
I think we should preserve the http status code besides the message, so that developers using the library can react to any specific http response scenarios they want; lets not take this information away from them.